### ISSUE TITLE
pylance randomly stopped recognizing used imports 

### ISSUE THREAD

**Issue by shawyan-t** [microsoft/pylance-release#4142]
Very simple example using a library I am importing:

```
import os
import spotipy
from spotipy.oauth2 import SpotifyClientCredentials

id = os.getenv('SPOTIFY_CLIENT_ID')
secret = os.getenv('SPOTIFY_CLIENT_SECRET')

creds = SpotifyClientCredentials(client_id=id, client_secret=secret)
spotOb = spotipy.Spotify(client_credentials_manager=creds)
```

I know this is not a spotipy issue as it has seemingly disabled the yellow recognizer in VSCODE (the color scheme for when an import is recognized and used) for all my imports

Now, even when accessed, it will display as white:
![image](https://user-images.githubusercontent.com/83904366/227505330-f64aa950-f5e6-4994-92c4-bf0433e9df1b.png)

noticeably, it will recognize other imports like time, making me think that perhaps it does not recognize my pip installs. In any case, the program this is in works just fine, same as before. The problem is the pylance linter. 

What occured prior to the error was creating a `__init__.py` file in a subdirectory within a project I am working on, so that I could access a file from a subdirectory. Ever since I have made that change, this bug has arisen and not left. I have since deleted the `__init__.py` files and the bug is still active. Any ideas on how I can fix this?

**Comment by debonte** [comment]
@shawyan-t, based on the image, Pylance is not reporting any diagnostics on the imports (ex. "Import spotipy could not be resolved"). Is that correct? Are you seeing any errors when trying to use symbols from these imported modules?

Btw, I'm not sure what the dashed lines are under some of the imports (ex. `time`). Is that from an extension? Or theme?

Assuming there are no errors related to the white imports, I wonder if this is a semantic highlighting issue. Could you try enabling "Developer: Inspect Editor Tokens and Scopes" via the command palette (Ctrl+Shift+P)? Then click on one of the miscolored import names and see what "semantic token type" is reported.

For the green module names, I would expect to see a token type of `module`. What do you see for the white ones?

![image](https://user-images.githubusercontent.com/12820898/227638679-ce75cadb-e472-406c-a57c-0cc2c85ec39a.png)

You can disable this feature by pressing `Esc` or executing the command via the command palette again.

**Comment by niushuqing123** [comment]
I have the same situation. Reinstalling vscode won't fix it either

**Comment by floxay** [comment]
I'm experiencing the same.
![image](https://user-images.githubusercontent.com/57007485/227861633-c15b79fe-f543-4373-8ef1-5083583ad15d.png)

After adding site-packages folder to Pylance's Extra Paths setting it seems suggestions and syntax highlighting are working as expected but tokenization(?) does not;
![image](https://user-images.githubusercontent.com/57007485/227862159-7d29a065-6832-4029-8718-fce7d456970f.png)

Pylance v2023.3.30
Python v2023.4.1
Python 3.11.2 on Windows 11 (22621.1413)

#4149 is most likely the same issue.

**Comment by floxay** [comment]
Update:
Went through older (pre)releases in VSC, `Pylance v2023.3.21` is the last that seems to work for third-party modules without manually adding site-packages to extra paths.

Logs:
[log-pylance_v2023.3.21.log](https://github.com/microsoft/pylance-release/files/11086234/log-pylance_v2023.3.21.log)
[log-pylance_v2023.3.22.log](https://github.com/microsoft/pylance-release/files/11086233/log-pylance_v2023.3.22.log)

`Pylance v2023.3.21`:
![image](https://user-images.githubusercontent.com/57007485/228149167-c2203a03-29c4-4f75-a2d9-5aa1753ae1f2.png)

`Pylance v2023.3.22`:
![image](https://user-images.githubusercontent.com/57007485/228149397-25e781d5-55b2-46ea-96de-65486d293772.png)

**Comment by debonte** [comment]
Thanks @floxay for tracking down the version where this behavior changed. I am able to repro this now and am investigating.

**Comment by phantasm** [comment]
@debonte I'm observing another similar issue with this - "Go To Definition (F12)" stopped working when I switch from v2023.3.21 to .22 while using python.analysis.extraPaths to inlcude all the needed codebase as I did before.
![image](https://user-images.githubusercontent.com/369127/228835151-a825e2f5-983a-4fae-a587-6053402704ff.png)

**Comment by shawyan-t** [comment]
> @shawyan-t, based on the image, Pylance is not reporting any diagnostics on the imports (ex. "Import spotipy could not be resolved"). Is that correct? Are you seeing any errors when trying to use symbols from these imported modules?
> 
> Btw, I'm not sure what the dashed lines are under some of the imports (ex. `time`). Is that from an extension? Or theme?
> 
> Assuming there are no errors related to the white imports, I wonder if this is a semantic highlighting issue. Could you try enabling "Developer: Inspect Editor Tokens and Scopes" via the command palette (Ctrl+Shift+P)? Then click on one of the miscolored import names and see what "semantic token type" is reported.
> 
> For the green module names, I would expect to see a token type of `module`. What do you see for the white ones?
> 
> ![image](https://user-images.githubusercontent.com/12820898/227638679-ce75cadb-e472-406c-a57c-0cc2c85ec39a.png)
> 
> You can disable this feature by pressing `Esc` or executing the command via the command palette again.


Sorry for the late reply, here is a screenshot of the Output after enabling "Developer: Inspect Editor Tokens and Scopes"

![develop](https://user-images.githubusercontent.com/83904366/229464456-48334dad-8a68-4507-a919-6a0aab9eb723.png)

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.4.11, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023411-5-April-2023-PreRelease)
